lib/tasker.js

```diff
@@ -0,0 +1,150 @@
+/* global Zinnia */
+
+import { ActivityState } from './activity-state.js'
+import { encodeHex } from '../vendor/deno-deps.js'
+import { assertOkResponse, assertRedirectResponse } from './http-assertions.js'
+import { getRandomnessForSparkRound } from './drand-client.js'
+import { assertEquals, assertInstanceOf } from 'zinnia:assert'
+
+/** @typedef {{cid: string; minerId: string;}} RetrievalTask */
+/** @typedef {RetrievalTask & { key: string}} KeyedRetrievalTask */
+
+export class Tasker {
+  #lastRoundUrl
+  /** @type {Task[]} */
+  #remainingRoundTasks
+  #fetch
+  #activity
+
+  /**
+   * @param {object} args
+   * @param {globalThis.fetch} args.fetch
+   * @param {ActivityState} args.activityState
+   */
+  constructor ({
+    fetch = globalThis.fetch,
+    activityState = new ActivityState()
+  } = {}) {
+    this.#fetch = fetch
+    this.#activity = activityState
+
+    this.maxTasksPerRound = 360
+
+    // TODO: persist these two values across module restarts
```

Comment:
What is the downside of not persisting this? We will always think a new round started when Spark starts?

Reply:
Added code comment